### ISSUE TITLE
iTead Sonoff ZBMini-L not leaving pairing mode after pairing

### ISSUE THREAD

**Issue by veitw** [zigpy/zha-device-handlers#1841]
**Describe the bug**
ZHA reports iTead Sonoff ZBMini-L to have successfully paired, but ZBMini-L does not leave pairing mode and tries to pair over and over again. While the device is still re-pairing, it can be controlled by ZHA (toggle switch, call identify, query cluster attributes, ...), but after the coordinator leaves pairing mode, the connection is lost.

**To Reproduce**
1. Configure ZHA.
2. Click on 'New device'.
3. Put ZBMini-L into pairing mode by holding its button for 5 seconds until green LED flashes at ~1 Hz.
4. Watch ZHA configuring the device over and over again, although it states pairing is finished and device can now be used.

**Expected behavior**
After pairing, ZBMini-L should leave pairing mode (green LED steadily on or off depending on relay state) and device being available for control.

<details>

<summary>Device signature</summary>

```yaml
{
  "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.EndDevice: 2>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress: 128>, manufacturer_code=4742, maximum_buffer_size=80, maximum_incoming_transfer_size=160, server_mask=11264, maximum_outgoing_transfer_size=160, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=True, *is_full_function_device=False, *is_mains_powered=False, *is_receiver_on_when_idle=False, *is_router=False, *is_security_capable=False)",
  "endpoints": {
    "1": {
      "profile_id": 260,
      "device_type": "0x0002",
      "in_clusters": [
        "0x0000",
        "0x0003",
        "0x0004",
        "0x0005",
        "0x0006",
        "0x0007",
        "0xfc57",
        "0xfca0"
      ],
      "out_clusters": [
        "0x0019"
      ]
    }
  },
  "manufacturer": "SONOFF",
  "model": "ZBMINI-L",
  "class": "zigpy.device.Device"
}
```

</details>

<details>

<summary>Diagnostic information</summary>

```yaml
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2022.10.4",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.10.5",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Europe/Berlin",
    "os_name": "Linux",
    "os_version": "5.15.72",
    "supervisor": "2022.10.0",
    "host_os": "Home Assistant OS 9.2",
    "docker_version": "20.10.17",
    "chassis": "vm",
    "run_as_root": true
  },
  "custom_components": {
    "worlds_air_quality_index": {
      "version": "0.3.3",
      "requirements": []
    },
    "hacs": {
      "version": "1.27.2",
      "requirements": [
        "aiogithubapi>=22.2.4"
      ]
    },
    "edge_tts": {
      "version": "0.0.1",
      "requirements": [
        "edge-tts==5.0.1"
      ]
    },
    "lovelace_gen": {
      "version": "0.1.1",
      "requirements": [
        "jinja2"
      ]
    },
    "browser_mod": {
      "version": "1.5.3",
      "requirements": []
    },
    "xiaomi_miio_fan": {
      "version": "2022.8.0.0",
      "requirements": [
        "construct==2.10.56",
        "python-miio>=0.5.12"
      ]
    },
    "anycubic_wifi": {
      "version": "1",
      "requirements": [
        "uart-wifi==0.2.1"
      ]
    },
    "zha_toolkit": {
      "version": "v0.8.16",
      "requirements": []
    },
    "rademacher": {
      "version": "1.9.3",
      "requirements": [
        "pyrademacher==0.6.6"
      ]
    },
    "mass": {
      "version": "2022.8.4",
      "requirements": [
        "music-assistant==1.8.7"
      ]
    }
  },
  "integration_manifest": {
    "domain": "zha",
    "name": "Zigbee Home Automation",
    "config_flow": true,
    "documentation": "https://www.home-assistant.io/integrations/zha",
    "requirements": [
      "bellows==0.34.2",
      "pyserial==3.5",
      "pyserial-asyncio==0.6",
      "zha-quirks==0.0.82",
      "zigpy-deconz==0.19.0",
      "zigpy==0.51.3",
      "zigpy-xbee==0.16.2",
      "zigpy-zigate==0.10.2",
      "zigpy-znp==0.9.1"
    ],
    "usb": [
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*2652*",
        "known_devices": [
          "slae.sh cc2652rb stick"
        ]
      },
      {
        "vid": "1A86",
        "pid": "55D4",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus v2"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*zigstar*",
        "known_devices": [
          "ZigStar Coordinators"
        ]
      },
      {
        "vid": "1CF1",
        "pid": "0030",
        "description": "*conbee*",
        "known_devices": [
          "Conbee II"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8A2A",
        "description": "*zigbee*",
        "known_devices": [
          "Nortek HUSBZB-1"
        ]
      },
      {
        "vid": "0403",
        "pid": "6015",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate+"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8B34",
        "description": "*bv 2010/10*",
        "known_devices": [
          "Bitron Video AV2010/10"
        ]
      }
    ],
    "codeowners": [
      "@dmulcahey",
      "@adminiuga",
      "@puddly"
    ],
    "zeroconf": [
      {
        "type": "_esphomelib._tcp.local.",
        "name": "tube*"
      },
      {
        "type": "_zigate-zigbee-gateway._tcp.local.",
        "name": "*zigate*"
      },
      {
        "type": "_zigstar_gw._tcp.local.",
        "name": "*zigstar*"
      }
    ],
    "dependencies": [
      "file_upload"
    ],
    "after_dependencies": [
      "onboarding",
      "usb",
      "zeroconf"
    ],
    "iot_class": "local_polling",
    "loggers": [
      "aiosqlite",
      "bellows",
      "crccheck",
      "pure_pcapy3",
      "zhaquirks",
      "zigpy",
      "zigpy_deconz",
      "zigpy_xbee",
      "zigpy_zigate",
      "zigpy_znp"
    ],
    "is_built_in": true
  },
  "data": {
    "ieee": "**REDACTED**",
    "nwk": 37593,
    "manufacturer": "SONOFF",
    "model": "ZBMINI-L",
    "name": "SONOFF ZBMINI-L",
    "quirk_applied": false,
    "quirk_class": "zigpy.device.Device",
    "manufacturer_code": 4742,
    "power_source": "Battery or Unknown",
    "lqi": 112,
    "rssi": -72,
    "last_seen": "2022-10-19T20:42:31",
    "available": true,
    "device_type": "EndDevice",
    "signature": {
      "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.EndDevice: 2>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress: 128>, manufacturer_code=4742, maximum_buffer_size=80, maximum_incoming_transfer_size=160, server_mask=11264, maximum_outgoing_transfer_size=160, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=True, *is_full_function_device=False, *is_mains_powered=False, *is_receiver_on_when_idle=False, *is_router=False, *is_security_capable=False)",
      "endpoints": {
        "1": {
          "profile_id": 260,
          "device_type": "0x0002",
          "in_clusters": [
            "0x0000",
            "0x0003",
            "0x0004",
            "0x0005",
            "0x0006",
            "0x0007",
            "0xfc57",
            "0xfca0"
          ],
          "out_clusters": [
            "0x0019"
          ]
        }
      }
    },
    "active_coordinator": false,
    "entities": [
      {
        "entity_id": "button.sonoff_zbmini_l_identifybutton",
        "name": "SONOFF ZBMINI-L"
      },
      {
        "entity_id": "select.sonoff_zbmini_l_startuponoffselect",
        "name": "SONOFF ZBMINI-L"
      },
      {
        "entity_id": "switch.sonoff_zbmini_l_switch",
        "name": "SONOFF ZBMINI-L"
      }
    ],
    "neighbors": [],
    "endpoint_names": [
      {
        "name": "ON_OFF_OUTPUT"
      }
    ],
    "user_given_name": null,
    "device_reg_id": "65a8a8e37055d7e0a6296db564f4eb4d",
    "area_id": null,
    "cluster_details": {
      "1": {
        "device_type": {
          "name": "ON_OFF_OUTPUT",
          "id": 2
        },
        "profile_id": 260,
        "in_clusters": {
          "0x0000": {
            "endpoint_attribute": "basic",
            "attributes": {
              "0x0004": {
                "attribute_name": "manufacturer",
                "value": "SONOFF"
              },
              "0x0005": {
                "attribute_name": "model",
                "value": "ZBMINI-L"
              }
            },
            "unsupported_attributes": {}
          },
          "0x0003": {
            "endpoint_attribute": "identify",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0004": {
            "endpoint_attribute": "groups",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0005": {
            "endpoint_attribute": "scenes",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0006": {
            "endpoint_attribute": "on_off",
            "attributes": {
              "0x0000": {
                "attribute_name": "on_off",
                "value": 0
              },
              "0x4003": {
                "attribute_name": "start_up_on_off",
                "value": 0
              }
            },
            "unsupported_attributes": {}
          },
          "0x0007": {
            "endpoint_attribute": "on_off_config",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0xfc57": {
            "endpoint_attribute": "manufacturer_specific",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0xfca0": {
            "endpoint_attribute": "manufacturer_specific",
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {
          "0x0019": {
            "endpoint_attribute": "ota",
            "attributes": {},
            "unsupported_attributes": {}
          }
        }
      }
    }
  }
}
```

</details>

<details>
<summary>Additional logs</summary>

```
Paste any additional debug logs here.
Don't remove the extra line breaks outside the ``` marks.
```

</details>

**Additional context**
All devices are on the latest software
* ZBMini-L runs firmware 1.1.1
* ZBDongle-E runs firmware 6.10.3-v1.0.1
* Home Assistant is at version 2022.10.4, running on HAOS 9.2

**Comment by veitw** [comment]
@Hedda Thank you very much for these hints. I have updated the description according to the bug report template.

I also have enabled debug logging as described and performed the pairing, but within these few minutes this produced 15k+ log lines including quite personal information that should not belong here. Do you have any further hints on what to grep from that log that is actually required for debugging such an issue, so I can filter out and redact these lines?

The method of contantly pressing the button while pairing is something I already tried, without success. Also I do not think this is a low energy/power saving issue, as due to its capacitors the device even continues to pair several times for up to about a minute even after being disconnected from power (breaker off, bulb unscrewed or plug pulled).

**Comment by veitw** [comment]
I found this in the forums:
https://community.home-assistant.io/t/endless-pairing-loops-for-some-devices/370602

It describes exactly the issue that I have, just for Innr and Ikea Zigbee LED bulbs instead of the Sonoff ZBMini-L.

Unfortunately, the workaround described (switching off/powering off the device after first successful pairing) does not work -- I tried this about 15 times today.

**Comment by orestesmas** [comment]
For your information, I have 3 of these devices working in my house and I'm not experiencing the issue you report.

I'm using a Conbee II coordinator with ZHA, as people you mentiones in your last message.

Nevertheless, I have other issues with these ZBMini-L devices: One of them is controlling a LED light in the porch, and I set up to turn on the light at sunset. Well, this works ok but I noticed that after a while (variable between 30 and 90 minutes approx.) the switch disconnects itself. I'm totally confused about that, because the logs doesn't show anything strange. I've to investigate this in depth, but I've read that these "no neutral required" devices need a load of around 3-4W minimum to work properly, and perhaps my LED bulb isn't an strong enough load and for that reason the relay disconnects.

I don't know if this behavior may be related to your problem, but I say it here just in case...

**Comment by veitw** [comment]
@orestesmas : Thanks for these infos. Are your ZBMini-L on firmware version 1.1.1, too?

Regarding the minimum load, the ZBMini-L is very humble compared to other no-neutral actuators, iTead even advertises that it does not require a bypass/capacitor being installed for low loads. In fact this is also true for my device, it works regardless whether only a single LED bulb is driven or an old 40W incandescent light bulb. I also measured the device to consume less than 0.5W (which is the lower limit my power meter does measure accurately) chained with either bulb.

**Comment by stperin** [comment]
Hi,

I have same issue on ZMINI-L firmware version 1.1.1

Did somebody found any solution?
Is there any alternative firmware to install to try to fix this?

**Comment by MattWestb** [comment]
@veitw Your EZSP version is one of the best and is not having problem joining "normal" devices (some extreme strange devices is all coordinators is having problems with).

Sonoff is having firmware problems with some new devices but i dont knowing is its related with this device and Z2M have digging deep for finding the problem and itead have being involved for workaround (not having the network open for joining that Z2M is having = not Zigbee 3 compliant and large security risk) but the work around is not working with ZHA then ZHA is only open the network for limited time  as shall being made for good Zigbee 3 coordinators.

https://github.com/Koenkk/zigbee2mqtt/issues/10282

I don have the device so i dont knowing what is happening but if some can sniffing one paring i can look if the coordinator and the device is doing all OK or if one part is not.

**Comment by thomasfr** [comment]
I can also confirm the behaviour. I am using Home Assistant Yellow (ezsp), bought 2 ZB Mini-L, both with the same issue. 
Pairing Mode (green led flashes), Home Assistant can find it, initializes, and says that the configuration is done and the device is ready to be used, while the green led keeps on flashing. When i change to the Devices section of Home Assistant it sometimes shows as available and sometimes it works to switch it on/off 2-3 times, but shortly after the logs say that the device is not available anymore.

**Comment by MattWestb** [comment]
Servus @thomasfr.
Do you have the possibility sniffing the paring with Yellow ?

If not I can doing it if can having one device for some days (time i cant guarantee ten its little chaotic in private for the moment).

Im in W1110 so shall being easy arranging form other all of Wien.

**Comment by thomasfr** [comment]
Oh servus! Nice to meet someone from vienna here :) 
Sure, i can.... if you can tell me how.

**Comment by MattWestb** [comment]
Im pinging you from my GITHUB for more info (I deleting my e-mail address later sent so its floating around).

I think its easy only some U3 stations and little walking :-))

**Comment by thomasfr** [comment]
I am not sure what you mean. 
How do sniff the pairing on home assistant?

**Comment by MattWestb** [comment]
I have pinging you in one new issue in my github in the message is my email but deleted in the issue now.
Better explaining way i have time problems for the moment.

For sniffing you need one CC2532 with sniff firmware or one normal EZSP coordinator but i can doing it for you.

**Comment by thomasfr** [comment]
I sent you an E-Mail

**Comment by MattWestb** [comment]
I have getting it and replayed.

Software for sniffing with EZSP https://github.com/zsmartsystems/com.zsmartsystems.zigbee.sniffer
that can being tricky getting it working (i think some modules or somthing is missing but i have it working on my laptop) plus wireshark.
All is explained in the link also links to guides and examples.

**Comment by Hedda** [comment]
Also see https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html#with-cc2531

**Comment by etienn01** [comment]
@veitw @stperin @thomasfr did you try with your ZigBee dongle on an USB extension cord? It fixed the problem for me with my ConBee II.

**Comment by veitw** [comment]
Hi @etienn01,

thanks for that hint. My Sonoff ZB Dongle Plus V2 (ZBDongle-E) is already offset from my HA box using a 3m USB 2.0 extension cord, is also connected to a USB 2.0 port instead of USB 3.x, and the HA box is wired by Ethernet with WiFi and BT disabled. 

Best regards,
// Veit

**Comment by SalN85** [comment]
Today I reported the same issue, it happened aftear i replaced my old cc2531 with a brand new sonoff usb dogle 3.0 (ZBDongle-E).
Note. In my network there is another Zbimi-l the works fine
Does anyone has solved the issue?

**Comment by roygeraets** [comment]
Hi Guys, 

I've got the same issue, ZHA and Z2MQTT. Same ZBDongle-E.
With Zigbee2MQTT it is possible to downgrade the firmware with the "Force" tag. 

But the Device itself does not allow the downgrade functionality.

Hope Itead will come with some decent firmware changes. I'm going to return my ZBMini-L devices as this is not the solution I'm hoping for.

**Comment by veitw** [comment]
As I also had problems with ZHA regarding my Osram/Ledvance A60 bulbs (in another ticket), which restart over and over again until eventually becoming stable after sometimes hours (which was quite annoying esp. at night in the bedroom, when after a short power outage the light switched on and on again every few seconds for hours), I switched to Z2M and the problems were instantly gone, both regarding my ZBMini-Ls as well as my Osram/Ledvance bulbs. It just worked as expected and I am so happy now.

All my ZBMini-L work well in Z2M with both the ZBDongle-E and also a new ZBDongle-P (CC2652P) that I bought as EZSP support is still experimental in Z2M and EZSP coordinator backup is not supported yet.

But I am still able to test for any updates as the ZBDongle-E is still connected to ZHA and I have one unused ZBMini-L left, though I have switched the ZBDongle-E to firmware version 7 to try ZHA+OpenThread in parallel but I could still switch back to firmware version 6 for testing until I get my first Matter-Thread devices. I have not yet tried the ZBMini-L with firmware version 7 though...

I read in a thread that people were successful pairing ZBMini-L after they paired a ZBDongle-P in router mode with Koenkk's router firmware and told ZHA to pair the problematic devices via the ZBDongle-P instead of the coordinator. I had not tried this as I read this after I already switched to Z2M with ZBDongle-P, but I had no success trying to pair the ZBMini-L via Ledvance bulbs and some Tuya plugs.

**Comment by roygeraets** [comment]
@veitw Can you post a link of ZBDongle-E firmware version 7.
The official version is 6.10.3 https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/Router

Would like to check if this maybe fixes the issue. 

> But I am still able to test for any updates as the ZBDongle-E is still connected to ZHA and I have one unused ZBMini-L left, though I have switched the ZBDongle-E to firmware version 7 to try ZHA+OpenThread in parallel but I could still switch back to firmware version 6 for testing until I get my first Matter-Thread devices. I have not yet tried the ZBMini-L with firmware version 7 though...

**Comment by veitw** [comment]
@roygeraets : This is the MultiPAN firmware that I installed:
https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle-NoSigned/RCPMultiPAN

This is not an official firmware, nor is it even signed, but I do not think that there is anything special about the ZBDongle-E that would prevent a generic EFR32MG21 firmware from working properly.

As I have no more Zigbee devices paired to ZHA, I cannot tell whether it actually works, I can only tell that ZHA says EZSP protocol is up, OpenThread seems to handle this well and that Z2M (stable) does not work with it beyond initial step of pairing as it does not support EZSP v7 yet. Did not try Z2M (dev) though.

Be warned that I am positive that you will have to re-pair devices or restore a coordinator backup when upgrading to EZSP v7 MultiPAN.

**Comment by MattWestb** [comment]
The xps1989 is / was from Sonoff devs but its not updated then the dev have moving on.
Signing is only for unhacked Sonoff ZBB and the unsignes is for the rest Sonofs EZSP.

Latest RCP / GSDK 7.20 is having one bug with direct children but its working good i have 30 mixed devices up and running with thread parallel running tuya MG21 and IKEA MG1P modules with ZHA. Then Silabs is fixing the bug HA is going Open thread with boarder router and matter support parallel with Zigbee on the same chip = all in.

**Comment by kylehowells** [comment]
Chiming in to say I've also been experiencing this with `Aqara smart wall switch H1 EU (with neutral, single rocker)`  [`WS-EUK03`](https://www.zigbee2mqtt.io/devices/WS-EUK03.html) device.

It pairs almost instantly and is controllable in ZHA HA, but the wall switch itself stays flashing it's LED in pairing mode and doesn't seem to think it is connected.
It works is is controllable, but every few seconds the device reboots, and then after about 5-10mins it exists pairing mode and shows the red "failed to pair" led color instead, and is no longer discoverable in HA.

It seems like the device is pairing, but then it isn't being sent whatever signal the device is expecting to tell it that it has worked.

**Comment by MauriceFlipse-eneco** [comment]
Same problem here

Zigbee Sonoff dongle 3.0
ESMLFzm_w6_TW by AwoX
